core/src/main/java/com/google/zxing/oned/OneDimensionalCodeWriter.java

```diff
@@ -90,6 +91,7 @@ private static BitMatrix renderResult(boolean[] code, int width, int height, int
     return output;
   }
 
+  private static final Pattern NUMERIC = Pattern.compile("[0-9]+");
```

Comment:
Can you write a simple `checkNumeric` method here or something and then just call it in each of the subclasses? that way the invocation isn't duplicated.